ocaml/xapi/storage_access.ml

```diff
@@ -65,6 +65,33 @@ let find_content ~__context ?sr name =
        || (vdi_rec.API.vDI_location = name) (* PR-1255 *)
     ) all
 
+let vdi_info_of_vdi_rec __context vdi_rec =
+  let content_id =
+    if List.mem_assoc "content_id" vdi_rec.API.vDI_other_config
+    then List.assoc "content_id" vdi_rec.API.vDI_other_config
+    else vdi_rec.API.vDI_location (* PR-1255 *)
```

Comment:
Could this be more efficient as try-with statement? i.e. `try List.assoc [...] with _ -> vdi_rec.API.vDI_location`

Reply:
Yes, that's safer, I think Jon also prefers that style, but we should  catch the `Not_found` exception. Didn't spot it as I just moved around existing code.